packages/navigator/src/browser/navigator-contribution.ts

```diff
@@ -207,4 +245,60 @@ export class FileNavigatorContribution extends AbstractViewContribution<FileNavi
             this.selectWidgetFileNode(this.shell.currentWidget);
         }
     }
+
+    protected async removeFolderFromWorkspace(): Promise<void> {
+        const uris = this.getSelectedNodes()
+            .filter(node => WorkspaceRootNode.is(node, this.workspaceService.numberOfRoots))
+            .map(rootFolder => {
+                const rootFolderNode = rootFolder as FileStatNode;
+                return new URI(rootFolderNode.uri.toString());
+            });
+
+        if (uris.length > 0) {
+            const messageContainer = document.createElement('div');
+            messageContainer.textContent = 'Remove the following folders from workspace? (note: nothing will be erased from disk)';
+            const list = document.createElement('ul');
+            list.style.listStyleType = 'none';
+            uris.forEach(u => {
+                const listItem = document.createElement('li');
+                listItem.textContent = u.displayName;
+                list.appendChild(listItem);
+            });
+            messageContainer.appendChild(list);
+            const dialog = new ConfirmDialog({
+                title: 'Remove Folder from Workspace',
+                msg: messageContainer
+            });
+            if (await dialog.open()) {
+                await this.workspaceService.removeRoots(uris);
+            }
+            return;
+        } else {
+            throw new Error(`Invalid workspace root URIs. Expected at least one existing directory location. URI: ${uris}.`);
```

Comment:
a command should be disabled if there is no matching uris, please look like `rename` or `delete` commands are implemented

Reply:
Why do we need the empty array as part of the error message?

Reply:
@akosyakov I agree that using `UriAwareCommandHandler` would result in simpler and more readable code. But i doubt if those two commands should be moved to `workspace-commands`.

Yes, they are related to workspace, but their enable/disable states are dependent on the navigator-tree info (e.g., is the selected node a `WorkspaceRootNode` etc). after being moved to the workspace package, they wouldn't have access to the required navigator-tree info.